### PR TITLE
Add shop directive

### PR DESCRIPTION
**Pull request by oterral**
[Test](https://mf-geoadmin3.dev.bgdi.ch/teo_shop/?lang=fr&topic=dev&bgLayer=ch.swisstopo.pixelkarte-farbe&X=193500.00&Y=799000.00&dev3d=true&zoom=1&layers_opacity=0.75&layers=ch.swisstopo.lk50-papierkarte-zusammensetzung.metadata)

### CODE REVIEW

**Comment by oterral** [comment]
@pauloamado [Test](https://mf-geoadmin3.dev.bgdi.ch/teo_shop/?lang=fr&topic=dev&bgLayer=ch.swisstopo.pixelkarte-farbe&X=193500.00&Y=799000.00&dev3d=true&zoom=1&layers_opacity=0.75&layers=ch.swisstopo.lk50-papierkarte-zusammensetzung.metadata)

**Comment by pauloamado** [comment]
Nice Job.
I will come back to you with 2 infos:
1. where to display the currency (CHF) 
2. first part of final test URL (http://www.toposhop.ch/#/) for the Shop dispatcher (handling the viewer request)
(http://www.toposhop.ch/#/de/dispatcher?layers=ch.swisstopo.pixelkarte-lk50.metadata&ch.swisstopo.pixelkarte-lk50.metadata=247)

**Comment by p1d1d1** [comment]
@oterral please organise the integrated shop layers (for paper maps) at least in swisstopo topic under "Paper Maps".
Planned deadline is 30.11.2015: it would be ok when ready during the next week.

PS: I confirm point 1) in Paulo's comment

File: src/components/shop/ShopDirective.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          return;
+        }
+
+        scope.orderTypes = layerConfig.shop;
+        scope.orderType = scope.orderTypes[0];
+
+        // {mapsheet,commune,district,canton,rectangle,whole}
+        // Order a mapsheet
+        scope.orderMapsheet = function() {
+          window.open('http://www.toposhop.ch/#/' + gaLang.get() +
```
Comment (cedricmoullet) [inline src/components/shop/ShopDirective.js]: Shouldn't we have several targets (dev,int, prod) ?

File: src/components/shop/ShopDirective.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        };
+        scope.chooseOrderType = function(ifScope) {
+          scope.orderType = ifScope.orderType;
+          var cap = scope.orderType.charAt(0).toUpperCase() +
+              scope.orderType.substr(1).toLowerCase();
+          var func = scope['order' + cap];
+          if (typeof func == 'function') {
+            func();
+          } else {
+            $window.console.log('order' + cap + ' not supported yet !');
```
Comment (cedricmoullet) [inline src/components/shop/ShopDirective.js]: console ?

File: src/components/shop/ShopDirective.js (hunk 1 above)
Comment (pauloamado) [inline src/components/shop/ShopDirective.js]: Currently, we don't have any test environment @swisstopo. This is a pending issue,for C2C.
We may use http://shop-shopbgdi.openshift.puzzle.ch but this URL will change later.
Example:
1)
http://shop-shopbgdi.openshift.puzzle.ch/#/de/dispatcher?layers=ch.swisstopo.pixelkarte-lk25.metadata&ch.swisstopo.pixelkarte-lk25.metadata=1011
will add 1011: Beggingen(lk25) to the basket and set the shop language to de.
2)
http://shop-shopbgdi.openshift.puzzle.ch/#/fr/dispatcher?layers=ch.swisstopo.pixelkarte-lk50.metadata&ch.swisstopo.pixelkarte-lk50.metadata=205
will add 205: Schaffhausen (lk50) to the basket and set the shop language to fr.

**Comment by gjn** [comment]
If catalog of a topic should be adapted, please use the normal process. Catalog is defined by IKP.

**Comment by oterral** [comment]
jenkins test

**Comment by p1d1d1** [comment]
@oterral there is one step missing. Please see Specifications UC 5.1.1 step 5.
Clicking on "Bestellen" will zoom to the selected object and open a confirmation window. Link to the Shop is in the confirmation window.

Thanks for updating.

**Comment by p1d1d1** [comment]
@oterral in "Object Information" windows the tootip highlight should be also around the "Order" button.
Thanks for updating

File: src/components/shop/ShopDirective.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+           scope.showConfirm = showConfirm;
+        };
+        // {mapsheet,commune,district,canton,rectangle,whole}
+        // Order a mapsheet
+        scope.goToShop = function() {
+          if (winShop) {
+            winShop.focus();
+          }
+          //var shopUrl = 'https://shop-shopbgdi.openshift.puzzle.ch/#/';
+          var shopUrl = 'https://shop-bgdi.dev.bgdi.ch/';
```
Comment (gjn) [inline src/components/shop/ShopDirective.js]: URL needs adaption for go live

**Comment by gjn** [comment]
@oterral Can you rebase and squash for tomorrows release?

**Comment by oterral** [comment]
sure.

File: Makefile [hunk 4]
```diff
@@ -400,7 +404,8 @@ prd/index.html: src/index.mako.html \
 	    ${HTMLMIN_CMD} \
 	    .build-artefacts/last-api-url \
 	    .build-artefacts/last-mapproxy-url \
-	    .build-artefacts/last-apache-base-path \
+	    .build-artefacts/last-shop-url \
+			.build-artefacts/last-apache-base-path \
```
Comment (gjn) [inline Makefile]: some ident problems in the make file.

File: rc_prod [hunk 5]
```diff
@@ -2,6 +2,7 @@ export KEEP_VERSION=true
 export DEPLOY_TARGET=prod
 export API_URL=//api3.geo.admin.ch
 export MAPPROXY_URL=//wmts{s}.geo.admin.ch
+export SHOP_URL=//shop-bgdi.prod.bgdi.ch
```
Comment (gjn) [inline rc_prod]: Official address, no?

Comment (oterral) [inline rc_prod]: @pauloamado is there an official address for production ?

Comment (pauloamado) [inline rc_prod]: official address is https://beta.shop.swisstopo.ch

File: Makefile [hunk 6]
```diff
@@ -400,6 +404,7 @@ prd/index.html: src/index.mako.html \
 	    ${HTMLMIN_CMD} \
 	    .build-artefacts/last-api-url \
 	    .build-artefacts/last-mapproxy-url \
+	    .build-artefacts/last-shop-url \
```
Comment (loicgasser) [inline Makefile]: minor and not directly related to this PR but we doing:
```
.build-artefacts/last-api-url \
.build-artefacts/last-mapproxy-url \
.build-artefacts/last-shop-url \
.build-artefacts/last-apache-base-path \
```

6 times, we could maybe factorize this rule.

Comment (oterral) [inline Makefile]: I've seen, will do it in the refactor make branch

**Comment by oterral** [comment]
prod url changed

**Pull request merged**